packages/ui/src/utils/steps.ts

```diff
@@ -1,5 +1,71 @@
 import type { Execute, RelayChain } from '@relayprotocol/relay-sdk'
-import type { Token } from '../types/index.js'
+import type { Token, LinkedWallet } from '../types/index.js'
+import { NormalizedWalletName } from '../constants/walletCompatibility.js'
+
+/**
+ * Get display name for wallet from linkedWallets array using current address
+ */
+const getWalletDisplayName = (
+  currentAddress?: string,
+  linkedWallets?: LinkedWallet[]
+): string => {
+  if (!currentAddress || !linkedWallets) {
+    return 'your wallet'
+  }
+
+  const linkedWallet = linkedWallets.find(
+    (linkedWallet) =>
+      currentAddress ===
+        (linkedWallet.vmType === 'evm'
+          ? linkedWallet.address.toLowerCase()
+          : linkedWallet.address) || linkedWallet.address === currentAddress
+  )
+
+  if (!linkedWallet?.connector) {
+    return 'your wallet'
+  }
+
+  // Use normalized wallet name if available, otherwise use connector directly
+  const normalizedName =
+    NormalizedWalletName[linkedWallet.connector] ?? linkedWallet.connector
+
+  // Capitalize first letter for display
+  return normalizedName.charAt(0).toUpperCase() + normalizedName.slice(1)
+}
+
+/**
+ * Replace generic wallet action text with wallet-specific text
+ */
+const customizeWalletActionText = (
+  action: string,
+  walletDisplayName: string
+): string => {
+  console.log(walletDisplayName)
+  // Replace "Confirm transaction in your wallet" with "Confirm in [WalletName]"
+  if (action === 'Confirm transaction in your wallet') {
```

Comment:
It'll be less brittle if it's based on the step id:
https://github.com/reservoirprotocol/solver/blob/86fb8d695d0c9fdfb3e90170fe9407146b516109/packages/cross-chain-solver/src/api/execute/quote/v1.ts#L376